expression/builtin_math.go

```diff
@@ -988,19 +989,23 @@ func (c *randFunctionClass) getFunction(ctx sessionctx.Context, args []Expressio
 
 type builtinRandSig struct {
 	baseBuiltinFunc
+	mu      *sync.Mutex
```

Comment:
All the projectWorkers will battle for this Mutex?

Reply:
Yes, and this is the only way to make it correct.